src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -707,6 +742,30 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
             }
         }
 
+        private BoundExpression BindDiscardedExpression(
```

Comment:
Consider returning narrower type, since we know it (`BoundDiscardedExpression`).